src/common/legacy/src/convert_function_to_cnn_network.cpp

```diff
@@ -1914,13 +1914,14 @@ void convertFunctionToICNNNetwork(const std::shared_ptr<const ::ngraph::Function
 
         // Copy runtime info attributes from Nodes to CNNLayers if they have std::string type
         for (const auto &rt : rt_info) {
-            if (rt.second.is<std::string>()) {
+            try {
                 auto str_attr = rt.second.as<std::string>();
                 if (details::CaselessEq<std::string>()(rt.first, "affinity")) {
                     cnnLayer->affinity = str_attr;
                 } else {
                     cnnLayer->params[rt.first] = str_attr;
                 }
+            } catch (...) {
             }
```

Comment:
why to change it ?

Reply:
> why to change it ?

Because rt attribute can be serializable but not string.

Reply:
Are you sure?
just checked Any::as() impl:

`template <class T>`
`T& as() & {`
`  type_check(typeid(decay_t<T>));`
`  return *static_cast<decay_t<T>*>(addressof());`
`}`

looks like `as<T>()` requires underlying type to be exactly `T`.

Reply:
> Are you sure? just checked Any::as() impl:
> 
> `template <class T>` `T& as() & {` ` type_check(typeid(decay_t<T>));` ` return *static_cast<decay_t<T>*>(addressof());` `}`
> 
> looks like `as<T>()` requires underlying type to be exactly `T`.

The following code:

            if (rt.second.is<std::string>()) {
                auto str_attr = rt.second.as<std::string>();
                if (details::CaselessEq<std::string>()(rt.first, "affinity")) {
                    cnnLayer->affinity = str_attr;
                } else {
                    cnnLayer->params[rt.first] = str_attr;
                }
            } else {
                std::cout << "IS NOT STRING " << rt.first << std::endl;
            }
            
gives:
```
IS NOT STRING keep_constants
IS NOT STRING can_be_folded
IS NOT STRING fused_names_0
IS NOT STRING gna_transpose_fusable_0
```

in the output and wrong tests results

Reply:
what test gives wrong results here?,
what do you mean by test results (eg. accuracy/some other assertion failure)
could you provide output from the test?

are you sure that `gna_transpose_fusable_0` param is copied by `cnnLayer->params[rt.first] = str_attr;` ? in your approach?
I am concerned that rt.second.as<std::string>() would throw exception here

Reply:
> what test gives wrong results here?, what do you mean by test results (eg. accuracy/some other assertion failure) could you provide output from the test?
> 
> are you sure that `gna_transpose_fusable_0` param is copied by `cnnLayer->params[rt.first] = str_attr;` ? in your approach? I am concerned that rt.second.asstd::string() would throw exception here

I just demonstrate that `rt.second.is<std::string>()` gives `false` even if attribute is serializable using `rt.second.as<std::string>()`. 

`gna_transpose_fusable_0` is copied with my changes.

Reply:
@kbruniec `as` doesn't require to be exactly T:
```
    template <class T>
    const T& as() const {
        return const_cast<Any*>(this)->as<T>();
    }
```

Reply:
You are right! I was looking at wrong as() impl.